plugin/pkg/global/resourcereferencemanager/admission_test.go

```diff
@@ -600,6 +603,84 @@ var _ = Describe("resourcereferencemanager", func() {
 					},
 				})))
 			})
+
+			It("should allow specifying a namespace which is not in use (create)", func() {
+				project.Spec.Namespace = pointer.StringPtr("garden-foo")
+				projectCopy := project.DeepCopy()
+				projectCopy.Name = "project-2"
+				projectCopy.Spec.Namespace = pointer.StringPtr("garden-bar")
+				gardenCoreInformerFactory.Core().InternalVersion().Projects().Informer().GetStore().Add(&projectCopy)
```

Comment:
This call actually errors because you are adding a pointer to `projectCopy` (which is already a `*core.Project`).
The test still compiles as the parameter is an `interface{}`, but the call errors and there will be no object in the store. 
But the test is therefore incorrect and will still pass, because you ignore the returned error.

I know, we do that in a lot of tests and IMHO we should stop ignoring errors returned by calls like this and always wrap such setup methods in `Expect(...).To(Succeed())`. Otherwise we may end up with incorrect tests that will pass.

Can you do that for all new test cases you added here?

```suggestion
				Expect(gardenCoreInformerFactory.Core().InternalVersion().Projects().Informer().GetStore().Add(projectCopy)).To(Succeed())
```

Reply:
addressed with #2075